### ISSUE TITLE
Compiler query will fail if compiler does not provide any system includes

### ISSUE THREAD

**Issue by adaml-at-eargo** [microsoft/vscode-cpptools#9099]
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows 10
- VS Code Version: 1.65.2
- C/C++ Extension Version: v1.9.7
- Other extensions you installed (and if the issue persists after disabling them):
- If using SSH remote, specify OS of remote machine: not using SSH

Attempting to configure Intellisense for use with the SEGGER clang compiler (an arm cross-compiler) always falls back to MSVC Intellisense using `cl.exe`. The result is that code is not properly parsed by Intellisense, e.g., types from `stdint.h` such as `uint32_t` are not defined. 

**Steps to reproduce**

1. Configure C/++ Properties for clang-arm Intellisense using the SES port of `clang.exe` (see below). 
2. Configure `"C_Cpp.loggingLevel": "Debug"`
3. Open a simple C file, e.g., `test.c` (below)
5. View the C/C++ output log, note line `MSVC intelliSenseMode specified.  Configuring for compiler cl.exe.`
6. Note that `uint32_t` is not a defined type.
7. Also note that the Intellisense system header include paths (from log diagnostics) now include Windows Kit paths and not the SES include paths.

**Expected behavior**

SES Clang should be used by Intellisense, with SES include paths. Standard types should be properly parsed.

**Code sample and logs**

- Code sample
```
// test.c
#include <stdint.h>

uint32_t some_function(uint32_t arg)
{
    return arg * 2u;
}
```

- Configurations in `c_cpp_properties.json`
```
{
    "name": "SES",
    "defines": [ "FLOAT_ABI_HARD", "NRF52840_XXAA"  ],
    "includePath": [     
        "C:\\Program Files\\SEGGER\\SEGGER Embedded Studio for ARM 6.22a\\include\\**",                          
        "${workspaceFolder}/**" ],
    "compilerPath": "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang.exe",
    "compilerArgs": [ "-nostdinc", "--target=arm","-mcpu=cortex-m4","-mfloat-abi=hard" ],
    "cStandard": "c11",
    "cppStandard": "c++17",
    "intelliSenseMode": "clang-arm"
} 
```

- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette
```
-------- Diagnostics - 3/27/2022, 4:09:33 PM
Version: 1.9.7
Current Configuration:
{
    "name": "SES",
    "defines": [
        "FLOAT_ABI_HARD",
        "NRF52840_XXAA"
    ],
    "includePath": [
        "C:\\Program Files\\SEGGER\\SEGGER Embedded Studio for ARM 6.22a\\include\\**",
        "${workspaceFolder}/application/**",
        "${workspaceFolder}/sdk/**"
    ],
    "compilerPath": "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang.exe",
    "compilerArgs": [
        "-nostdinc",
        "--target=arm",
        "-mcpu=cortex-m4",
        "-mfloat-abi=hard"
    ],
    "cStandard": "c11",
    "cppStandard": "c++17",
    "intelliSenseMode": "clang-arm",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "C:\\Program Files\\SEGGER\\SEGGER Embedded Studio for ARM 6.22a\\include\\**",
            "${workspaceFolder}/application/**",
            "${workspaceFolder}/sdk/**",
            "${workspaceFolder}"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Users\Adam.Lins\dev\test\test.c ]:
    C:\Users\Adam.Lins\dev\test\test.c
Translation Unit Configurations:
[ C:\Users\Adam.Lins\dev\test\test.c ]:
    Process ID: 63648
    Memory Usage: 16 MB
    Compiler Path: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
    Includes:
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\BUILDTOOLS\VC\TOOLS\MSVC\14.29.30133\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.19041.0\CPPWINRT
    Defines:
        FLOAT_ABI_HARD
        NRF52840_XXAA
    Standard Version: c11
    IntelliSense Mode: windows-clang-arm
Total Memory Usage: 16 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 9244

------- Potential include path issues --------
Some headers exist in multiple locations. If IntelliSense is behaving incorrectly,
try adding one of the alternate paths to the "includePath" in your configuration in
c_cpp_properties.json to override the automatic path discovery for that header.

Using: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30133/include/stdint.h
    Alternative: "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/include"
```

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)
```
Attempting to get defaults from C compiler in "compilerPath" property: 'C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang'
Querying compiler for default C++ language standard using command line: "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang" -x c++ -E -dM nul
Detected language standard version: c++14
Querying compiler for default C language standard using command line: "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang" -x c -E -dM nul
Detected language standard version: c17
Querying compiler's default target using command line: "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang" -dumpmachine
Compiler returned default target value: x86_64-pc-windows-msvc

Compiler query command line: "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang" -nostdinc --target=arm -mcpu=cortex-m4 -mfloat-abi=hard -std=c11 -Wp,-v -E -dM -x c  -fno-blocks nul
MSVC intelliSenseMode specified.  Configuring for compiler cl.exe.
Shutting down IntelliSense server: C:\Users\Adam.Lins\dev\test\test.c
Attempting to get defaults from C++ compiler in "compilerPath" property: 'C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang'
Compiler query command line: "C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang" -nostdinc --target=arm -mcpu=cortex-m4 -mfloat-abi=hard -std=c++17 -Wp,-v -E -dM -x c++  -fno-blocks nul
MSVC intelliSenseMode specified.  Configuring for compiler cl.exe.
  Folder: C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2019/BUILDTOOLS/VC/TOOLS/MSVC/14.29.30133/INCLUDE/* will be indexed
  Folder: C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.19041.0/CPPWINRT/ will be indexed
  Folder: C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.19041.0/SHARED/ will be indexed
  Folder: C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.19041.0/UCRT/ will be indexed
  Folder: C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.19041.0/UM/ will be indexed
  Folder: C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.19041.0/WINRT/ will be indexed
  Folder: C:/PROGRAM FILES/SEGGER/SEGGER EMBEDDED STUDIO FOR ARM 6.22A/INCLUDE/ will be indexed
  Folder: C:/USERS/ADAM.LINS/DEV/TEST/ will be indexed
```

**Screenshots**

**Additional context**

Output from running clang.exe from a powershell terminal attached as `diagnostic_output.txt`:
```
& 'C:/Program Files/SEGGER/SEGGER Embedded Studio for ARM 6.22a/llvm/bin/clang' -nostdinc --target=arm -mcpu=cortex-m4 -mfloat-abi=hard -std=c11 `-Wp,-v -E -dM -x c  -fno-blocks nul 
```
[diagnostic_output.txt](https://github.com/microsoft/vscode-cpptools/files/8358718/diagnostic_output.txt)

SES invokes the compiler using a `cc1`-style interface. The built-in header paths don't match the actual location on disk, so `-nostdinc` and `-isystem` options are used. The `-isystem` option in the C/C++ configuration doesn't seem to be used by Intellisense during configuration.

**Comment by Colengms** [comment]
Hi @adaml-at-eargo .  Thanks for reporting this.  This is the first compiler we've encountered with this behavior.  The issue appears to be that no system paths are being returned, and the C/C++ extension is considering that an error.  It looks like no paths being returned is by-design for this compiler. (Using a `-isystem` manually causes that path to be included in what the compiler returns, and would work around the issue if the C/C++ extension were not also filtering the arg out as unnecessary for the query).

I have a fix in queue.  This should be addressed in our next Insiders build.

**Comment by sean-mcmanus** [comment]
This should be fixed with 1.10.2 (pre-release).

**Comment by adaml-at-eargo** [comment]
@sean-mcmanus , coincidentally I had started using the pre-release version a couple of days ago. It's working very well. Thank you all!
